tests/test_RadClass.py

```diff
@@ -98,7 +99,41 @@ def test_cache():
     # the resulting 1-hour observation should be:
     #   counts * integration / live-time
     expected = spectra * integration / test_data.livetime
-    results = np.genfromtxt('results.csv', delimiter=',')[1, 1:]
+    results = classifier.storage.to_numpy()[1]
     np.testing.assert_almost_equal(results, expected[0], decimal=2)
 
-    os.remove('results.csv')
+
+def test_start():
```

Comment:
I don’t think this test is complete because I don’t think you skip data in the cache?

you probably need a dummy data file where you can tell if the cache has been accessed properly with a non-zero start time

Reply:
I'm not sure I understand. The line `start_time = timestamps[integration]` starts at the next integration period and thus skips the first. I added a `num_results` variable to hopefully make it clearer. The `stride` & `integration` are constructed using this variable so that `len(classifier.storage)` in the absence of a start/stop time would be this number. Instead, because I use the above line of code, the first integration period is skipped and the last is skipped (because it would be an incomplete period, see `RadClass.py` ln 165). Therefore, the total number of result samples here should be `num_results-2`.

This assumes that the cache works properly, as tested in `test_cache`. Perhaps there is a better way to test this than checking for an expected number of results?

Reply:
My point was based on the fact that you were not originally updating `start_i` in `queue_file()`, which should have caused this test to fail.  The test wasn't failing, so it suggests that the test is not well formed.  It may relate to the fact that you are using a uniform spectrum, as we're discussing in #17

Reply:
You also never test the contents of the result of the integration, and just the size of the result.

Reply:
I have added an test of the resulting integration using the same method of calculation as the other tests.